selenium_tests/homepage/test_banners.py

```diff
@@ -0,0 +1,19 @@
+from selenium_tests.base import BaseSeleniumTests
+from home.factories import (
+    BannerIndexPageFactory,
+    BannerFactory,
+    ArticleFactory
+)
+
+class BannerSeleniumTests(BaseSeleniumTests):
+
+    def setUp(self):
+        self.setup_blank_site()
+        self.banner_index = BannerIndexPageFactory(parent=self.site.root_page)
+
+    def test_banner_is_present(self):
+        self.article01 = ArticleFactory(parent = self.site.root_page, title = 'article01')
+        self.banner01 = BannerFactory(parent = self.banner_index, title = 'banner01')        
```

Comment:
@sheralim012 I am trying to build a test which sets up some basic banners and then checks they are appearing/ performing as expected. Think I have understood how the banners work, but struggling to dynamically create the banners with code such that they appear on the homepage (so that I can then test them). Would appreciate if you had any thoughts on the most efficient way to get banners on the homepage. At the moment I just have factories to create the banner index and banner page but its creating the link to get them appearing on the homepage that I'm struggling with.

Reply:
HomePageBanner.objects.create(source=home_page, banner_page=banner_page)

for reference: https://github.com/unicef/iogt/blob/develop/iogt_content_migration/management/commands/load_v1_db.py#L1706